### PR TITLE
apply routes to resource

### PR DESCRIPTION
**Pull request by 928PJY**
#3091

### CODE REVIEW

File: src/docfx/build/Build.cs [hunk 1]
```diff
@@ -129,7 +129,7 @@ private static void BuildResource(Context context, Document file)
 
             if (file.Docset.Config.Output.CopyResources)
             {
-                context.Copy(file, file.FilePath);
+                context.Copy(file, Document.ApplyRoutes(file.FilePath, file.Docset.Config.Routes));
```
[the remark this answers is not in the document]
Comment (yufeih) [reply src/docfx/build/Build.cs]: also can we have a test case in config.yml

File: docs/specs/config.yml [hunk 2]
```diff
@@ -167,20 +173,28 @@ inputs:
...
   outputB/b.json:
+  outputC/c.gif:
   build.manifest:
 ---
 # latter routes rule takes precedence - 2
 inputs:
   docfx.yml: |
     routes:
       docs/a/b.md: outputB
+      docs/a/c.gif: outputC
```
Comment (yufeih) [inline docs/specs/config.yml]: remove this line, to check `docs/a` should include `docs/a/c.gif`

File: src/docfx/config/RouteConfig.cs [hunk 3]
```diff
@@ -28,6 +28,11 @@ public string GetOutputPath(string path)
         {
             Debug.Assert(!string.IsNullOrEmpty(path));
 
+            if (Source == "./")
```
Comment (yufeih) [inline src/docfx/config/RouteConfig.cs]: is this covered in test?

File: docs/specs/config.yml (hunk 2 above)
Comment (928PJY) [reply docs/specs/config.yml]: this line doesn't influent the check, but can also check the routes priority in resource

---
In reply to: [203285427](https://github.com/dotnet/docfx/pull/3092#discussion_r203285427) [](ancestors = 203285427)

**Pull request merged**